### ISSUE TITLE
apt fails to retrieve any new packages

### ISSUE THREAD

**Issue by i3anaan** [laradock/php-fpm#13]
This one for sure caused me a headache:
I wanted to install the php7.1 intl extension in our php-fpm container, however, instructions such as listed here were not working:
https://www.colinodell.com/blog/2016-12/installing-php-7-1

Another example failure:
`sudo docker run -u 0 [php-fpm container] apt-get install nano`

Whatever I tried, the container was unable to locate *any* package that was not already installed. The source list was fine, but after a long search, I found the culprit: `ls -la /var/lib/apt/lists/` showed nothing, the folder was empty. Running `apt-get update` did **not** restore this folder, it stayed empty.

Getting back to the code in this repository, the following line seemed to cause this issue:
[`&& rm -rf /var/lib/apt/lists/*`](https://github.com/laradock/php-fpm/blob/master/Dockerfile-71#L32)

Now is my experience with Debian and php-fpm limited, but this, disabling the functioning of `apt`, definitely does not seem like intended behaviour?

For now, our workaround will be to skip the laradock image and use the source form the dockerfile without the breaking line, but I would be very happy if this could be fixed.

**Comment by thorerik** [comment]
It's definitely on purpose, you can rebuild this by running `apt update`, since you say this did not restore the functionality, please provide the entire output of running the command.

**Comment by i3anaan** [comment]
Hey,

Here you go for a log detailing the issue:

```
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 ls -la /var/lib/apt/lists
[sudo] password for i3anaan: 
total 8
drwxr-xr-x 4 root root 4096 Apr 22 02:30 .
drwxr-xr-x 9 root root 4096 Apr 22 02:30 ..
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt update

WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Get:2 http://security.debian.org jessie/updates/main amd64 Packages [556 kB]
Ign http://deb.debian.org jessie InRelease
Get:3 http://deb.debian.org jessie-updates InRelease [145 kB]
Get:4 http://deb.debian.org jessie Release.gpg [2373 B]
Get:5 http://deb.debian.org jessie-updates/main amd64 Packages [17.8 kB]
Get:6 http://deb.debian.org jessie Release [148 kB]
Get:7 http://deb.debian.org jessie/main amd64 Packages [9063 kB]
Fetched 9996 kB in 5s (1855 kB/s)
Reading package lists...
Building dependency tree...
Reading state information...
36 packages can be upgraded. Run 'apt list --upgradable' to see them.
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 ls -la /var/lib/apt/lists
total 8
drwxr-xr-x 4 root root 4096 Apr 22 02:30 .
drwxr-xr-x 9 root root 4096 Apr 22 02:30 ..
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt-get updateGet:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Get:2 http://security.debian.org jessie/updates/main amd64 Packages [556 kB]
Ign http://deb.debian.org jessie InRelease
Get:3 http://deb.debian.org jessie-updates InRelease [145 kB]
Get:4 http://deb.debian.org jessie Release.gpg [2373 B]
Get:5 http://deb.debian.org jessie-updates/main amd64 Packages [17.8 kB]
Get:6 http://deb.debian.org jessie Release [148 kB]
Get:7 http://deb.debian.org jessie/main amd64 Packages [9063 kB]
Fetched 9996 kB in 5s (1851 kB/s)
Reading package lists...
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 ls -la /var/lib/apt/lists
total 8
drwxr-xr-x 4 root root 4096 Apr 22 02:30 .
drwxr-xr-x 9 root root 4096 Apr 22 02:30 ..
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt-get install nano
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package nano
i3anaan@i3anaan-Legion:~$ 
```

**Comment by i3anaan** [comment]
Another interesting case, it says there are 36 packages to upgrade, and you can use `apt list --upgradable` to show them, yet this shows none:

```
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt update

WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Get:2 http://security.debian.org jessie/updates/main amd64 Packages [556 kB]
Ign http://deb.debian.org jessie InRelease
Get:3 http://deb.debian.org jessie-updates InRelease [145 kB]
Get:4 http://deb.debian.org jessie Release.gpg [2373 B]
Get:5 http://deb.debian.org jessie Release [148 kB]
Get:6 http://deb.debian.org jessie-updates/main amd64 Packages [17.8 kB]
Get:7 http://deb.debian.org jessie/main amd64 Packages [9063 kB]
Fetched 9996 kB in 5s (1974 kB/s)
Reading package lists...
Building dependency tree...
Reading state information...
36 packages can be upgraded. Run 'apt list --upgradable' to see them.
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt list --upgradable

WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

Listing...
i3anaan@i3anaan-Legion:~$ 
```

**Comment by i3anaan** [comment]
Although not shown above, just to be sure I removed and redownloaded the container, which failed on installing nano just the same:

```
i3anaan@i3anaan-Legion:~$ sudo docker run laradock/php-fpm:1.4-71 apt-get install nano
Unable to find image 'laradock/php-fpm:1.4-71' locally
1.4-71: Pulling from laradock/php-fpm
Digest: sha256:4401240d81af58b8fab02d15377c5925528001eeeb8b5a878ab76e09b12a3939
Status: Downloaded newer image for laradock/php-fpm:1.4-71
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package nano
```

**Comment by i3anaan** [comment]
Tried to get a one-line command that shows the container is up-to-date and showcase the error, gives permission errors, could this be an indicator of what is going wrong?

```
i3anaan@i3anaan-Legion:~$ sudo docker run -u 0 laradock/php-fpm:1.4-71 whoami && apt-get update && apt-get install nano
Unable to find image 'laradock/php-fpm:1.4-71' locally
1.4-71: Pulling from laradock/php-fpm
Digest: sha256:4401240d81af58b8fab02d15377c5925528001eeeb8b5a878ab76e09b12a3939
Status: Downloaded newer image for laradock/php-fpm:1.4-71
root
Reading package lists... Done
W: chmod 0700 of directory /var/lib/apt/lists/partial failed - SetupAPTPartialDirectory (1: Operation not permitted)
E: Could not open lock file /var/lib/apt/lists/lock - open (13: Permission denied)
E: Unable to lock directory /var/lib/apt/lists/
W: Problem unlinking the file /var/cache/apt/pkgcache.bin - RemoveCaches (13: Permission denied)
W: Problem unlinking the file /var/cache/apt/srcpkgcache.bin - RemoveCaches (13: Permission denied)
```

**Comment by i3anaan** [comment]
Let me know if I can provide you with anything else that could help you, although I would say these logs should suffice :)

**Comment by i3anaan** [comment]
Okay,
First off: I realized I made a mistake in my reasoning previously, I have deleted those comments, I was incorrectly using docker causing while generating those logs: they were kind of useless after I found this out.

Secondly: I got the laradock/php-fpm container to update its packages and to install php-intl (my goal). So in that sense, this issue is most likely fixed and was caused by a user error on my side.
I will try and reproduce the incorrect result and hopefully find out what I did wrong before closing this issue.

Anyway, hope I did not waste too much of your time and thank you!

**Comment by i3anaan** [comment]
I am almost ashamed of what went wrong earlier:

1. I tried to install php-intl using commands that were intended for Ubuntu, clearly not working for Debian
2. I then most likely forgot to run `apt-update` causing me to believe `apt install` was broken
3. Removing the line I blamed in this issue had *no effect*
4. Misunderstanding of docker led me to further believe `apt update` was not working

All in all, I am happy I manage to resolve this mostly by myself, once again I hope it was mostly my own time I wasted on this, although I did learn some new stuff :)
